amundsen_application/static/js/components/AvatarLabelList/index.tsx

```diff
@@ -0,0 +1,215 @@
+
```

Comment:
I don't know how much it will simplify the logic, but it can still only be done if we leverage Redux correctly, since that's the only way to have one component edit a piece of state that another component is also displaying.

I think I may consider doing the rework of the tableMetadata state, and then update this PR.

Reply:
Yeah I wasn't sure how much simpler the split components would be given our current setup. Okay to revisit this later.

Reply:
I'll keep as is. A second use case for a  "+ Add" button that opens a modal will justify separating these out.